packages/client/lib/client.ts

```diff
@@ -106,8 +106,16 @@ export default class EthereumClient extends events.EventEmitter {
     if (this.started) {
       return false
     }
-    await Promise.all(this.config.servers.map((s) => s.start()))
-    await Promise.all(this.services.map((s) => s.start()))
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    Promise.all(this.services.map((s) => s.start()))
+      .then(() => {
+        return Promise.all(this.config.servers.map((s) => s.start()))
+      })
+      .then(() => {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        Promise.all(this.config.servers.map((s) => s.bootstrap()))
+      })
```

Comment:
would this achieve the same?

```suggestion
    await Promise.all(this.services.map((s) => s.start()))
    await Promise.all(this.config.servers.map((s) => s.start()))
    await Promise.all(this.config.servers.map((s) => s.bootstrap()))
```

Reply:
i believe libp2p runs its own bootstrap, so if the `bootstrap()` method is only available for `RlpxServer`, we could add a simple guard to the last line, something like: `if('bootstrap' in s) { s.bootstrap() }`

Reply:
No, this is just not working, this was my naive (is it?) approach as well on first try. You can test this I guess now also on `master` with `Aleut`. If you use this clean `await` chaining version of yours and do a `DEBUG=devp2p:* npm run client:start -- --network=aleut` run you should see no `Send GET_BLOCK_HEADERS` response from our side. If you run with this bad hacky code from my test fix you should see a response together with some follow-up processing (going wrong with "could not decode message BlockHeaders"):

![image](https://user-images.githubusercontent.com/931137/115984370-2099bd80-a5a7-11eb-9478-aa367f385a0d.png)

My Promise-feeling unfortunately is so bad that it seems that I don't get this right. Or maybe this just needs some stronger sequentiality with some artificial delays with a timer or so due to the networking stuff? Not sure if this makes sense but we can also try.

Anyhow: I am a bit stuck how to fix this.

Reply:
thanks for the info, ok I'll keep looking into it and report back, I am curious now too :)

Reply:
I found the problem, we needed to await the `Devp2pRLPx` initialization, fixed in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1222/commits/0297fd5093d21a0831ddab9d90e2471c4e2e9b42